client/core/trade.go

```diff
@@ -665,11 +667,37 @@ func (t *trackedTrade) isRedeemable(match *matchTracker) bool {
 	return false
 }
 
+// shouldBeginFindRedemption will be true if we are the Taker on this match,
+// we've broadcasted a swap, our swap has gotten the required confs, we've
+// not been notified of Maker's redeem and we've not refunded our swap.
+func (t *trackedTrade) shouldBeginFindRedemption(match *matchTracker) bool {
+	proof := &match.MetaData.Proof
+	swapCoinID := proof.TakerSwap
+	if match.failErr != nil || match.Match.Side != order.Taker || len(swapCoinID) == 0 ||
+		len(proof.MakerRedeem) > 0 || len(proof.RefundCoin) > 0 {
+		log.Tracef(
+			"Not finding redemption for match %v: side = %s, failErr = %v, TakerSwap = %v RefundCoin = %v",
+			match.id, match.Match.Side, match.failErr, proof.TakerSwap, proof.RefundCoin)
+		return false
+	}
+	if _, alreadyFinding := t.awaitingRedemption[swapCoinID.String()]; alreadyFinding {
```

Comment:
does this need a lock?

Reply:
The trade mutex is already locked before this method is called in trade.tick().